libs/pika/resource_partitioner/include/pika/resource_partitioner/partitioner_fwd.hpp

```diff
@@ -50,6 +49,25 @@ namespace pika::resource {
         mode_allow_dynamic_pools = 2
     };
 
+#if defined(PIKA_HAVE_MPI)
+    /// This enumeration describes polling pool creation modes,
+    /// the user may request a dedicated pool that can be used by pika::mpi
+    /// MPI pool completion flags are passed on the command line or via env vars
+    /// When pika+MPI is disabled, no mpi pool will be created, the polling mode
+    /// is ignored and using xxx_enable/xxx_disable has no effect
+    enum polling_pool_creation_mode
+    {
+        /// the user must create a pool if they wish to use one with mpi
+        mode_manual = 0,
+        /// if the completion mode requires it, a pool will be created at startup
+        mode_pika_decides = 1,
+        /// overrides command line flags/env vars - enables creation of a polling pool
+        mode_force_create = 2,
+        /// overrides command line flags/env vars - disables creation of a polling pool
+        mode_force_no_create = 3,
+    };
+#endif
+
```

Comment:
The enum doesn't mention anything about "MPI" but the docstrings keep mentioning "MPI". There's obviously the CUDA polling as well, which is not mentioned here. Either we:
- keep that out of this and if in the future we want a CUDA polling pool we duplicate this, but then the variables should be MPI-specific (i.e. have `mpi` in the name)
- or we leave the topic open and don't mention MPI here now, to leave it open so that CUDA polling could in the future be told to happen on the (one and only) polling pool

What do you think?

Reply:
Tjhat's exactly why I was being cryptic - I left MPI out of the actual name/flags, but mentioned it in the docs, because right now we only use it for mpi, but the first polling changes also moved cuda to the pool (that was reverted a long time ago), but we might want to try it again, now that stuff is more mature.
I'll rewrite the doc to be a bit more explicit about current use cases ...